apis/serving/v1beta1/inferenceservice_registry.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2021 IBM Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/kserve/modelmesh-serving/apis/serving/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// +kubebuilder:object:generate=false
+type InferenceServiceRegistry struct {
+	Client client.Client
+}
+
+func BuildPredictorWithBase(isvc *InferenceService, framework string, frameworkSpec *PredictorExtensionSpec) *v1alpha1.Predictor {
+	p := &v1alpha1.Predictor{}
+
+	p.Spec = v1alpha1.PredictorSpec{
+		Model: v1alpha1.Model{
+			Type: v1alpha1.ModelType{
+				Name: framework,
+			},
+		},
+	}
+
+	if frameworkSpec != nil && frameworkSpec.RuntimeVersion != nil {
+		p.Spec.Runtime = &v1alpha1.PredictorRuntime{
+			RuntimeRef: &v1alpha1.RuntimeRef{
+				Name: *frameworkSpec.RuntimeVersion,
+			},
+		}
+	}
+	return p
+}
+
+func (isvcr InferenceServiceRegistry) Get(ctx context.Context, nname types.NamespacedName) (*v1alpha1.Predictor, error) {
+	inferenceService := &InferenceService{}
+	err := isvcr.Client.Get(ctx, nname, inferenceService)
+
+	if err != nil {
+		return nil, err
+	}
+
+	framework, frameworkSpec := inferenceService.Spec.Predictor.GetPredictorFramework()
+	if frameworkSpec == nil {
+		return nil, errors.New("No valid InferenceService predictor framework found")
+	}
+
+	p := BuildPredictorWithBase(inferenceService, framework, frameworkSpec)
+
+	p.TypeMeta = inferenceService.TypeMeta
```

Comment:
Hmm I'm not sure we want to copy the TypeMeta, rather just set it to `v1.TypeMeta{Kind: "Predictor", APIVersion: "serving.kserve.io/v1"}` (though for practical purposes currently I don't think it makes a difference).

Reply:
I guess my intent here was to pass the TypeMeta through the Predictor so that `UpdateStatus` will get the original InferenceService TypeMeta, however, after more fiddling, seems like I can just completely remove these TypeMeta lines, and everything still seems to work fine. Perhaps fine to just remove?